File: 3034-number-of-subarrays-that-match-a-pattern-i/3034-number-of-subarrays-that-match-a-pattern-i.cpp
class Solution {
public:
    int countMatchingSubarrays(vector<int>& nums, vector<int>& pattern) {
        int ans=0;
        for(int i=0;i<nums.size()-1;i++){
             int k=0,j=i;
             bool flag=true;
            while((k<pattern.size())&&(j<nums.size()-1)){
                
                if(pattern[k]==1){
                    cout<<nums[j]<<" "<<nums[j+1]<<" ";
                    if(nums[j]>=nums[j+1]){
                     flag=false;  
                    }       
                }
                else if(pattern[k]==0){
                    
                    if(nums[j]!=nums[j+1]){
                        flag=false;  
                    }
                    
                }
                else if(pattern[k]==-1){
                    if(nums[j]<=nums[j+1]){
                        flag=false;
                    }
                }
                j++;
                k++;     
            }
           
            if(k==pattern.size()&&flag==true)
                ans++;
           
           
        }
        return ans;
    }
};

File: 3034-number-of-subarrays-that-match-a-pattern-i/3034-number-of-subarrays-that-match-a-pattern-i.java
class Solution {
    public int countMatchingSubarrays(int[] nums, int[] pattern) {
        int result = 0;
        for (int i = 0; i < nums.length - pattern.length; i++) {
            boolean matches = true;
            for (int j = 0; j < pattern.length; j++) {
                int index = i + j;                
                int num1 = nums[index];
                int num2 = nums[index + 1];
                if (num2 > num1 && pattern[j] != 1) {
                    matches = false;
                }
                if (num2 == num1 && pattern[j] != 0) {
                    matches = false;
                }
                if (num1 > num2 && pattern[j] != -1) {
                    matches = false;
                }
            }
            if (matches) {
                result++;
            }
        }
        return result;
    }
}